packages/providers/src/base.provider.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  camelCase,
+  capitalCase,
+  constantCase,
+  kebabCase,
+  pascalCase,
+  snakeCase,
+} from './utils/change-case';
+import { IOptions } from './utils/change-case/functions';
+import { deepmerge } from './utils/deepmerge.utils';
+import { WithPassthrough } from './utils/types';
+
+export enum CasingEnum {
+  CAMEL_CASE = 'camelCase',
+  PASCAL_CASE = 'PascalCase',
+  CAPITAL_CASE = 'CapitalCase',
+  SNAKE_CASE = 'snake_case',
+  KEBAB_CASE = 'kebabcase',
+  CONSTANT_CASE = 'CONSTANT_CASE',
+}
+
+type TransformOutput<T> = {
+  body: T;
+  headers: Record<string, string>;
+  query: Record<string, string>;
+};
+
+export abstract class BaseProvider {
+  protected casing: CasingEnum | undefined;
+
+  protected keyCaseObject: Record<string, string> = {};
+
+  protected transform<
+    T_Output = Record<string, unknown>,
+    T_Input = Record<string, unknown>,
+    T_Data = Record<string, unknown>
+  >(
+    bridgeProviderData: WithPassthrough<T_Input>,
+    triggerProviderData: T_Data
+  ): TransformOutput<T_Output> {
+    const {
+      _passthrough = {
+        body: {},
+        headers: {},
+        query: {},
```

Comment:
I think we need to add a fallback value for each of these properties. Without this fallback, if I specify a `_passthrough = { body: { key: 'value' } }`, the `headers` and `query` will not be populated with a default value and will be `undefined`.

Reply:
Should we just do a deepmerge of the fallback `_passthrough` and the provided `_passthrough` to handle these fallbacks?

Reply:
A yes I missed it for body lower down